main.cpp

```diff
@@ -402,6 +444,63 @@ IntType& IntType::divide(int rhs)
     return *this;
 }
 
+struct Point
+{
+    Point(const FloatType& fta, const FloatType& ftb)
+    {
+        FloatType xCopy = const_cast<FloatType&>(fta);
+        x = static_cast<float>(xCopy);
+        FloatType yCopy = const_cast<FloatType&>(ftb);
+        y = static_cast<float>(yCopy);
+    }
+    Point(const DoubleType& dta, const DoubleType& dtb)
+    {
+        x = static_cast<float>( const_cast<DoubleType&>(dta));
+        y = static_cast<float>( const_cast<DoubleType&>(dtb));
+    }
+    Point(const IntType& ita, const IntType& itb)
+    {
+        x = const_cast<IntType&>(ita);
+        y = const_cast<IntType&>(itb);
+    }
```

Comment:
Do not use `const_cast`.  Find another way to make this work.   
hint: what is missing from your conversion function?
- [ ] revise